src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -1209,34 +1235,48 @@ public void Range_Add_Success(Uri remoteServer)
             Assert.Equal(request.Headers["Range"], "bytes=1-5");
         }
 
-        [Theory, MemberData(nameof(EchoServers))]
-        public async Task GetResponseAsync_PostRequestStream_ContainsData(Uri remoteServer)
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task GetResponseAsync_PostRequestStream_ContainsData(bool useSsl)
         {
-            HttpWebRequest request = HttpWebRequest.CreateHttp(remoteServer);
-            request.Method = HttpMethod.Post.Method;
+            var options = new LoopbackServer.Options { UseSsl = useSsl };
 
-            using (Stream requestStream = await request.GetRequestStreamAsync())
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
             {
-                requestStream.Write(_requestBodyBytes, 0, _requestBodyBytes.Length);
-            }
+                HttpWebRequest request = WebRequest.CreateHttp(uri);
+                request.ServerCertificateValidationCallback = (a, b, c, d) => true;
+                request.Method = HttpMethod.Post.Method;
+                using (Stream requestStream = await request.GetRequestStreamAsync())
+                {
+                    requestStream.Write(_requestBodyBytes, 0, _requestBodyBytes.Length);
+                }
 
-            using (WebResponse response = await request.GetResponseAsync())
-            using (Stream myStream = response.GetResponseStream())
-            using (var sr = new StreamReader(myStream))
-            {
-                string strContent = sr.ReadToEnd();
-                Assert.True(strContent.Contains(RequestBody));
-            }
+                using (WebResponse response = await request.GetResponseAsync())
+                using (Stream myStream = response.GetResponseStream())
+                using (var sr = new StreamReader(myStream))
+                {
+                    string strContent = sr.ReadToEnd();
+                    Assert.True(strContent.Contains(RequestBody));
+                }
+            }, server => server.HandleRequestAsync(content: RequestBody), options);
```

Comment:
This isn't right.  The previous version was using an echo server, and by the client code checking that the response contained the request body, it was thus checking that the request body was correctly sent to the server (which then echo'd it).  This change is making it so that the server simply ignores the request body and instead simply sends back the same content, without verifying that's actually what it received.

Reply:
I agree with @stephentoub on this point. HttpWebRequest builds on HttpClient and we already have detailed tests for the HttpClient layer. So, we use these HttpWebRequest tests mainly to verify end-to-end functionality. And using an echo server helps satisfy that requirement.